nifi-nar-bundles/nifi-graph-bundle/nifi-graph-processors/src/main/java/org/apache/nifi/processors/graph/ExecuteGraphQueryRecord.java

```diff
@@ -0,0 +1,265 @@
+            writer.finishRecordSet();
+            long end = System.currentTimeMillis();
+            delta = (end - start) / 1000;
+            if (getLogger().isDebugEnabled()){
+                getLogger().debug(String.format("Took %s seconds.", delta));
+            }
+        } catch (Exception ex) {
+            getLogger().error("", ex);
+            failed = true;
+        } finally {
+            if (failed) {
+                session.remove(output);
+                session.getProvenanceReporter().route(input, REL_FAILURE);
+                session.transfer(input, REL_FAILURE);
+            } else {
+                input = session.putAttribute(input, GRAPH_OPERATION_TIME, String.valueOf(delta));
+                session.getProvenanceReporter().send(input, clientService.getTransitUrl(), delta*1000);
+                session.transfer(input, REL_SUCCESS);
+                FlowFile graph = session.create(input);
+                session.getProvenanceReporter().clone(input, graph);
+                boolean error = false;
+                try (OutputStream os = session.write(graph)) {
+                    String graphOutput = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(graphResponses);
+                    os.write(graphOutput.getBytes(StandardCharsets.UTF_8));
+                    session.getProvenanceReporter().modifyContent(graph);
+                } catch (Exception ex) {
+                    getLogger().error("", ex);
+                    error = true;
+                } finally {
+                    if (!error) {
+                        session.getProvenanceReporter().route(graph, REL_GRAPH);
+                        session.transfer(graph, REL_GRAPH);
+                    } else {
+                        session.remove(graph);
+                    }
+                }
+                if (errors.get() > 0) {
+                    session.getProvenanceReporter().route(output, REL_ERRORS);
+                    session.transfer(output, REL_ERRORS);
+                } else {
+                    session.remove(output);
+                }
+            }
+        }
+    }
+
+    private String getScript(PropertyDescriptor script, ProcessContext context, FlowFile input) {
+        String scriptValue = context.getProperty(script).evaluateAttributeExpressions(input).getValue();
+        return new StringBuilder()
+                .append(scriptValue)
+                .toString();
+    }
+
+    private List<Map<String, Object>> executeQuery(String recordScript, Map<String, Object> parameters) {
+        ObjectMapper mapper = new ObjectMapper();
+        List<Map<String, Object>> graphResponses = new ArrayList<>();
+        clientService.executeQuery(recordScript, parameters, (map, b) -> {
+            if (getLogger().isDebugEnabled()){
+                try {
+                    getLogger().debug(mapper.writerWithDefaultPrettyPrinter().writeValueAsString(map));
+                } catch (JsonProcessingException ex) {
```

Comment:
An error reporting would be helpful here.

Reply:
Added in an error reporting, though likely will rarely get executed.